### ISSUE TITLE
Adding newline in between frontmatter delimiters causes issues

### ISSUE THREAD

**Issue by aczw** [wooorm/markdown-tm-language#10]
Hi! Title says it all but when I try adding... any kind of newline in between the frontmatter delimiters (`---`), the syntax highlighting goes hay wire. I'm coming from MDX's VS Code extension. I've turned off all extensions except for MDX's and my theme.

- MDX extension v1.8.9
- VS Code 1.92.0

Without frontmatter:

![image](https://github.com/user-attachments/assets/2ef310ac-7a49-4663-aced-9ae725b011cd)

With frontmatter, but empty:

![image](https://github.com/user-attachments/assets/f707fe33-ea2b-4785-b3d7-de37a884c571)

With frontmatter _and_ a newline:

![image](https://github.com/user-attachments/assets/9924b5b6-8767-4f1d-946e-f6510aaf124f)

With frontmatter and actual values:

![image](https://github.com/user-attachments/assets/dd76975d-5426-4eec-a596-13e837b573da)

Still occurs when I turn my theme off as well:

![image](https://github.com/user-attachments/assets/0e66c6b9-2c1a-4c34-9b0b-03e85119c137)

Please let me know how I can help. Thank you!

**Comment by wooorm** [comment]
Not reproducible on the demo here: https://wooorm.com/markdown-tm-language/.

I also don’t see it locally in VS Code with MDX.

Perhaps report this to the YAML grammar you use with VS Code. Looks like they mess up when there’s a

**Comment by remcohaszing** [comment]
Interesting :thinking: 

I am unable to reproduce this in my own VSCode setup. However, I am able to reproduce this with all extensions disabled. Installing the [YAML](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) extension resolves it.

This means it’s a bug with the [builtin VSCode YAML grammar](https://github.com/microsoft/vscode/tree/main/extensions/yaml). The YAML extension uses [its own grammar](https://github.com/redhat-developer/vscode-yaml/blob/main/syntaxes/yaml.tmLanguage.json) which isn’t broken.

Both VSCode and the YAML extension pull the YAML grammar from external sources, but not the same source. So really this is an issue with the external YAML grammar that VSCode uses.

TL;DR: It’s an issue in VSCode (https://github.com/microsoft/vscode/issues/224862). As a workaround, install the [YAML](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) extension. It’s really good, you should use it anyway. :)

**Comment by aczw** [comment]
How bizarre! Installing the YAML extension did solve the issue, thank you.

**Comment by Firgrep** [comment]
Also had this issue and the YAML extension resolved it! Cheers!
